source/extensions/filters/http/basic_auth/BUILD

```diff
@@ -13,7 +13,7 @@ envoy_cc_library(
     name = "basic_auth_lib",
     srcs = ["basic_auth_filter.cc"],
     hdrs = ["basic_auth_filter.h"],
-    external_deps = ["ssl"],
+    external_deps = ["ssl", "bcrypt"],
```

Comment:
its not an external dep as defined here - i needs to be referenced as a normal dep using its `@name`

Reply:
Thanks @phlax for sharing, will do it

Reply:
just to add - you will most likely need to use `foreign_cc` to build it

Reply:
Please use `@com_github_hilch_bcrypt//:bcriypt_lib` directly. Maybe you also need to add a build file for the dependency because it doesn't use bazel.

Reply:
@wbpcode i dont get the idea of where to add this build file.
```
licenses(["notice"])  # BSD-4

cc_library(
    name = "bcrypt",
    hdrs = glob(["/Bcrypt.cpp/include/*.h"]),
    includes = ["Bcrypt.cpp/include"],
    visibility = ["//visibility:public"],
)
```

This seems to be the bazel config we need but not sure where to add

Is it possible for one of u to explain how does the `repository_locations.bzl` file work in this setup